etc/kayobe/ansible/octavia-amphora-image-build.yml

```diff
@@ -0,0 +1,95 @@
+---
+- name: Build an Octavia Amphora image
+  hosts: "{{ amphora_builder_group | default('seed') }}"
+  tasks:
+    - name: Ensure packages are installed
+      become: true
+      vars:
+        packages_for_os_family:
+          RedHat:
+            - debootstrap
+            - qemu-img
+            - git
+            - e2fsprogs
+            - policycoreutils-python-utils
+            - yum-utils
+          Debian:
+            - debootstrap
+            - qemu-utils
+            - git
+            - kpartx
+      package:
+        name: "{{ packages_for_os_family[ansible_facts.os_family] }}"
+      
+    - name: Create a temporary directory
+      tempfile:
+        state: directory
+      register: tempfile_result
+
+    - block:
+        - name: Check whether the image cache directory exists
+          stat:
+            path: "{{ image_cache_path }}"
+            get_md5: False
+            get_checksum: False
+            mime: False
+          register: image_cache_stat
+
+        - name: Ensure the image cache directory exists
+          file:
+            path: "{{ image_cache_path }}"
+            state: directory
+            owner: "{{ ansible_facts.user_uid }}"
+            group: "{{ ansible_facts.user_gid }}"
+          become: true
+          when: >-
+            not image_cache_stat.stat.exists or
+            not image_cache_stat.stat.writeable
+
+        - name: Set path facts
+          vars:
+            work_path: "{{ tempfile_result.path }}"
+          set_fact:
+            src_path: "{{ work_path }}/octavia"
+            venv_path: "{{ work_path }}/venv"
+            work_path: "{{ work_path }}"
+
+        - name: Clone Octavia source code
+          git:
+            depth: 1
+            dest: "{{ src_path }}"
+            repo: "https://opendev.org/openstack/octavia"
+            version: "{{ openstack_branch }}"
+
+        - name: Install diskimage-builder in a virtual environment
+          pip:
+            name: diskimage-builder
+            # NOTE: Ussuri-Wallably upper constraints all fail to build the image.
+            # This is just for diskimage-builder, and doesn't affect the version of amphora.
+            extra_args: "-c https://opendev.org/openstack/requirements/raw/branch/stable/xena/upper-constraints.txt"
```

Comment:
What about yoga?

Reply:
Shall I change back to `{{ pip_upper_constraints_file }}`?

Reply:
Added `amphora_dib_upper_constraints_file` that defaults to `pip_upper_constraints_file`